extensions-core/druid-catalog/src/main/java/org/apache/druid/catalog/http/CatalogListenerResource.java

```diff
@@ -0,0 +1,126 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.catalog.http;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.jaxrs.smile.SmileMediaTypes;
+import org.apache.druid.catalog.model.TableMetadata;
+import org.apache.druid.catalog.model.TableSpec;
+import org.apache.druid.catalog.sync.CatalogUpdateNotifier;
+import org.apache.druid.catalog.sync.MetadataCatalog.CatalogListener;
+import org.apache.druid.guice.annotations.Json;
+import org.apache.druid.guice.annotations.Smile;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.server.security.Access;
+import org.apache.druid.server.security.Action;
+import org.apache.druid.server.security.AuthorizationUtils;
+import org.apache.druid.server.security.AuthorizerMapper;
+import org.apache.druid.server.security.Resource;
+import org.apache.druid.server.security.ResourceAction;
+import org.apache.druid.server.security.ResourceType;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+@Path(CatalogListenerResource.BASE_URL)
+public class CatalogListenerResource
+{
+  public static final String BASE_URL = "/druid/broker/v1/catalog";
+  public static final String SYNC_URL = "/sync";
+
+  private final CatalogListener listener;
+  private final AuthorizerMapper authorizerMapper;
+  private final ObjectMapper smileMapper;
+  private final ObjectMapper jsonMapper;
+
+  @Inject
+  public CatalogListenerResource(
+      final CatalogListener listener,
+      @Smile final ObjectMapper smileMapper,
+      @Json final ObjectMapper jsonMapper,
+      final AuthorizerMapper authorizerMapper)
+  {
+    this.listener = listener;
+    this.authorizerMapper = authorizerMapper;
+    this.smileMapper = smileMapper;
+    this.jsonMapper = jsonMapper;
+  }
+
+  @POST
+  @Path(SYNC_URL)
+  @Consumes({MediaType.APPLICATION_JSON, SmileMediaTypes.APPLICATION_JACKSON_SMILE})
+  public Response syncTable(
+      final InputStream inputStream,
+      @Context final HttpServletRequest req)
+  {
+    Response resp = checkAuth(req);
+    if (resp != null) {
+      return resp;
+    }
+    final String reqContentType = req.getContentType();
+    final boolean isSmile = SmileMediaTypes.APPLICATION_JACKSON_SMILE.equals(reqContentType);
+    final ObjectMapper mapper = isSmile ? smileMapper : jsonMapper;
+    TableMetadata tableSpec;
+    try {
+      tableSpec = mapper.readValue(inputStream, TableMetadata.class);
+    }
+    catch (IOException e) {
+      return Response.serverError().entity(e.getMessage()).build();
```

Comment:
Think the error here could be improved. Maybe the error should contain an identifier for the table that was attempted to be synced, and the operation type (delete, update, etc), along with the exception message itself.